plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,253 @@
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.ml.cluster.DiscoveryNodeHelper;
+import org.opensearch.ml.common.MLTask;
+import org.opensearch.ml.common.MLTaskState;
+import org.opensearch.ml.common.MLTaskType;
+import org.opensearch.ml.common.transport.load.MLLoadModelAction;
+import org.opensearch.ml.common.transport.load.MLLoadModelRequest;
+import org.opensearch.ml.utils.MLNodeUtils;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Manager class for ML models and nodes. It contains ML model auto reload operations etc.
+ */
+@Log4j2
+public class MLModelAutoReLoader {
+    private final Client client;
+    private final ClusterService clusterService;
+    private final NamedXContentRegistry xContentRegistry;
+    private final DiscoveryNodeHelper nodeHelper;
+    private volatile Boolean enableAutoReLoadModel;
+
+    /**
+     * constructor method， init all the params necessary for model auto reloading
+     * @param clusterService
+     * @param client
+     * @param xContentRegistry
+     * @param nodeHelper
+     * @param settings
+     */
+    public MLModelAutoReLoader(
+        ClusterService clusterService,
+        Client client,
+        NamedXContentRegistry xContentRegistry,
+        DiscoveryNodeHelper nodeHelper,
+        Settings settings
+    ) {
+        this.clusterService = clusterService;
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+        this.nodeHelper = nodeHelper;
+
+        enableAutoReLoadModel = ML_COMMONS_MODEL_AUTO_RELOAD_ENABLE.get(settings);
+        clusterService
+            .getClusterSettings()
+            .addSettingsUpdateConsumer(ML_COMMONS_MODEL_AUTO_RELOAD_ENABLE, it -> enableAutoReLoadModel = it);
+    }
+
+    /**
+     * the main method: model auto reloading
+     */
+    public void autoReLoadModel() {
+        log.debug("enableAutoReLoadModel {} ", enableAutoReLoadModel);
+
+        // if we don't need to reload automatically, just return without doing anything
+        if (!enableAutoReLoadModel) {
+            return;
+        }
+
+        // At opensearch startup, get local node id, if not ml node,we ignored, just return without doing anything
+        String localNodeId = clusterService.localNode().getId();
+        if (!MLNodeUtils.isMLNode(nodeHelper.getNode(localNodeId))) {
+            return;
+        }
+
+        // According to the node id to get retry times, if more than the maxi retry times, don't need to retry
+        // that the number of unsuccessful reload has reached the maximum number of times, do not need to reload
+        Integer reTryTimes = getReTryTimes(localNodeId);
```

Comment:
Basically we use encapsulated type in class fields, and in methods we usually use primitive type to avoid null checking logic. The retryTimes is totally controlled by ourself, so it makes sense to use primitive type instead of encapsulated type.